### DISCUSSION TITLE
Nodes indices and Coordinates in Updated Mesh Format

### DISCUSSION THREAD

**Discussion by JulienKh** [Concrete-Chrono-Development/chrono-preprocessor#7] [Q&A] [answered]
I have 2 questions regarding the new mesh files:
1-How can I obtain the indices of the particles/nodes for each facet (previously denoted as n1 and n2 in the old facet files)? I believe they are now in the tet file, but clarification on the order is needed.
2-Where can I locate fxN, fyN, fzN (the coordinates for nodes of each facet)? They used to be present in the old facet files.

**Comment by mtroemner** [comment, marked as answer]
To your questions:

1. In the file `...-data-facets.dat` the first value on each line (labeled as `tet`) is the tet number where that facet resides. In the file `...-data-tets.dat` each tet is provided with the list of nodes with which it is comprised. Finally, the facets follow a pre-specified tet-node connectivity (i.e. left column facet 0 connected by node 0 and 1; second column facet 1 is connected by node 0 and 1;... right column facet 11 is connected by node 2 and 3):
    `tetn1 = [0,0,0,0,0,0,1,1,1,1,2,2]`
    `tetn2 = [1,1,2,2,3,3,2,2,3,3,3,3]`
With this information the n1 and n2 can be fully-reconstructed as needed.

2. In the file `...-data-facets.dat` the second, third, and fourth values of each line (labeled as `IDx`, `IDy`, and `IDz`) are facet vertex ID numbers. These correspond to the coordinates in the file `...-data-facetsVertices.dat`. So in example, if the file `...-data-facets.dat` has a first line that looks like: `0 0 1 2 1.72011128 1.078061968 5.134209899 1.086083926 20.64495912 0.8750093554 0 0.484106009 -0.4789421256 0.1456705436 0.8656757667 -0.0705199855 -0.9893331556 0.1274630885 0`; then the facet vertex ID values are `0 1 2`, and we would pull out the first three lines (i.e. first three coordinates) from the file `...-data-facetsVertices.dat` for that facet.

Feel free to follow-up if anything is unclear. Otherwise if you feel this has answered your question, please mark as answered. Thanks!
